### PR TITLE
[Aqara] Aqara Smart Dimmer Controller T1 Pro

### PR DESCRIPTION
**Pull request by sbkimaqara**
Aqara Smart Dimmer Controller T1 Pro

### CODE REVIEW

File: drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+local function do_configure(self, device)
+  device:send(ColorControl.commands.MoveToColorTemperature(device, 200, 0x0000))
+  device:configure()
+  do_refresh(self, device)
+end
+
+local function device_added(driver, device, event)
+  device:emit_event(capabilities.switch.switch.on())
+  device:emit_event(capabilities.switchLevel.level(100))
+  device:emit_event(capabilities.colorTemperature.colorTemperature(5000))
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua:80]: shouldn't events for all of these be generated organically by the device responding to the `refresh` command?

File: drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      device:send(Level.attributes.OnTransitionTime:write(device, raw_value))
+    end
+
+    if device.preferences[lightFadeOutTimeInSec.ID] ~= nil and
+        device.preferences[lightFadeOutTimeInSec.ID] ~= args.old_st_store.preferences[lightFadeOutTimeInSec.ID] then
+      local raw_value = device.preferences[lightFadeOutTimeInSec.ID] -- seconds
+      raw_value = raw_value * 10 -- unit: 100ms
+      device:send(Level.attributes.OffTransitionTime:write(device, raw_value))
+    end
+  end
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua:68]: you repeat this pattern a lot, perhaps creating a mapping of all of the preferences to a function to call if the value has changed would reduce some of the boilerplate you're using in your drivers?

e.g. (in your base driver function, with switching based on model name, see https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zwave-sensor/src/preferences.lua for a similar idea):

```
LIGHT = {
  MODELS = { mfr = ..., model = ...},
  PREFERENCES = {
    lightFadeInTimeInSec = function(device, new_value) ... end,
    ...
  }
},
...
```

then you could just use one generic configure lifecycle handler across all your devices and adding devices with new preferences would be simpler.

File: drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua (hunk 1 above)
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua:80]: OK, I will delete the code.

File: drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua (hunk 2 above)
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/aqara-light/init.lua:68]: OK, I will update the code.

**Review by greens — COMMENTED** [review]
Almost. I was suggesting you use the map and the `device_info_changed` function into a separate file so that all of your drivers can use the same pattern.

**Comment by sbkimaqara** [comment]
Is it correct to write the code by creating a new file in "/drivers/SmartThings/zigbee-switch/src/preferences.lua" path?

**Comment by greens** [comment]
Yes. I think that would work.

**Comment by sbkimaqara** [comment]
> Yes. I think that would work.

OK, I updated. Please review my code.

**Review by greens — APPROVED** [review]

**Review by ctowns — APPROVED** [review]

**Pull request merged**